encryption/encryption.cc

```diff
@@ -265,6 +266,15 @@ Status KeyManagedEncryptedEnv::NewSequentialFile(
     case EncryptionMethod::kAES192_CTR:
     case EncryptionMethod::kAES256_CTR:
       s = encrypted_env_->NewSequentialFile(fname, result, options);
+      // Hack: sometimes the current file may be read as an encrypted
+      // file by mistakes(when upgrading from TiKV <= v5.0.0-rc).
+      if (s.ok() && IsCurrentFile(fname)) {
+        if (!IsValidCurrentFile(std::move(*result))) {
```

Comment:
Add a comment explaining that on certain occasion a plaintext current file is written before stale encryption info is cleaned up.